### PR TITLE
Update pydoctor

### PR DESCRIPTION
**Pull request by tristanlatr**
## Scope and purpose

- Update pydoctor to at least 24.11.0. 
- Fixes the "Go to the latest version of this document." links. 
- Use the new pydoctor option --html-base-url. 
- Remove google analytics from the API docs.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,7 +62,7 @@ test = [
 # release scripts and process.
 dev-release = [
     "towncrier ~= 23.6",
-    "pydoctor ~= 23.9.0",
+    "pydoctor >= 24.11.0",
```
Comment (adiroiban) [inline pyproject.toml]: For our dev dependencies, we do a strict pin, trying to reduce the posibility of breaking the CI due to future pydoctor updates.

```suggestion
    "pydoctor ~= 24.11.1",
```

File: src/twisted/python/_pydoctortemplates/subheader.html [hunk 2]
```diff
@@ -5,25 +5,5 @@
...
-      link.href = window.location.pathname.replace(/\/\d+\.\d+\.\d+\/api\//, '/current/api/');
-      <!-- And make it visible -->
-      var container = document.getElementById('current-docs-container');
-      container.style.display = "";
-      delete link;
-      delete container;
-    }
-  </script>
-  
+  <script src="stable-link.js" type="text/javascript"></script>
```
Comment (adiroiban) [inline src/twisted/python/_pydoctortemplates/subheader.html:8]: Happy to see the google analytics removed.

-------

Just asking 

Does moving the script to a separate file improves the performance ?

**Review by adiroiban — APPROVED** [review]
Thanks for the changes

I have reviewed the result at https://twisted--12384.org.readthedocs.build/en/12384/api/twisted.logger.html


I prefer to pin the pydoctor dependency.

----------

This needs to have a `src/twisted/newsfragments/12384.misc` to let our release notes processing tools know that it should not expect a public release for this PR

-------

Other than that, only minor comments

While reviewing, I have observed something that is not specific to this PR. The issue is also found on `trunk`.

The source links for the `stable` branch are alwasy to `HEAD`, instead of linking to the branch.

Most probably this is not a `pydoctor` issue, but an issue with the way we build the api docs

File: src/twisted/python/_pydoctortemplates/subheader.html (hunk 2 above)
Comment (tristanlatr) [reply src/twisted/python/_pydoctortemplates/subheader.html:8]: I don’t believe it particularly improves performance, no

**Comment by tristanlatr** [comment]
Are we good here ?

**Comment by tristanlatr** [comment]
> The source links for the stable branch are alwasy to HEAD, instead of linking to the branch.

This is true only for the tox environment, for the production Twisted uses correct paths: https://github.com/twisted/twisted/blob/a0175bb49cb563b43bea8be7db065db7a366f292/docs/conf.py#L195

**Comment by adiroiban** [comment]
> This needs to have a src/twisted/newsfragments/12384.misc to let our release notes processing tools know that it should not expect a public release for this PR

Other than that, I think that this is ready to merge.

Thanks for the update!

**Pull request merged**